<a href="https://colab.research.google.com/github/Mathrn/matheus_rocha/blob/main/curso_em_video_lista_python.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lista de exercicios do curso em video - professor guanabara
python 0-100


basicamente, estou iniciando pela aula. nela o professor aborda tipos primitivos de dados.
são eles:
int - numeros inteiros
bool - verdadeiro ou falso
float - numeros reais
str - textos

In [ ]:
#3) crie um programa que leia dois numeros e mostre a soma entre eles
n1 = int(input("digite um numero:"))
n2 = int(input("digite um numero: "))
s = n1+n2
print(f'a soma dos numeros informados é {s}')

digite um numero:12
digite um numero: 35
a soma dos numeros informados é 47


In [ ]:
#4) faça um programa que leia algo pelo teclado e mostre na tela o seu tipo primitivo
# e todas as informacoes sobre ele.
algo  = (input("digite um numero inteiro: "))
print(f"tipo primitivo do que foi digitado: {type(algo)}")
print("o que foi digitado é numerico? ",algo.isnumeric())
print(f'todos os caracteres são letras? {algo.isalpha()}')
print(f'todas as letras são minusculas? {algo.islower()}')





digite um numero inteiro: 35rA
tipo primitivo do que foi digitado: <class 'str'>
o que foi digitado é numerico?  False
todos os caracteres são letras? False
todas as letras são minusculas? False


AULA 5- OPERADORES ARITMETICOS

In [ ]:
'''5- Faça um programa que leia um número Inteiro e mostre na tela o seu sucessor e seu antecessor.'''
num =int(input("digite um numero: "))
ant = num-1
suc = num+1
print(f'o numero informado foi {num} o antecessor é {ant} e o seu sucessor é {suc}')


digite um numero: 12
o numero infromado foi 12 o antecessor é 11 e o seu sucessor é 13


In [ ]:
'''6-Crie um algoritmo que leia um número e mostre o seu dobro, triplo e raiz quadrada.'''
num = float(input("digite um numero: "))
print(f'o numero digitado foi {num} \n o seu dobro é {num*2}, \n triplo {num*3:.3f} \n e sua raiz é {num**(0.5):.3f}')

#conhecimento novo : o comando \n quebra o print e coloca o conteudo dele em outra linha

digite um numero: 1589
o numero digitado foi 1589.0 
 o seu dobro é 3178.0, 
 triplo 4767.000 
 e sua raiz é 39.862


In [ ]:
'''7 -  Desenvolva um programa que leia as duas notas de um aluno, calcule e mostre a sua média.'''
nota1 = float(input("digite sua nota:"))
nota2 = float(input("digite sua nota: "))
med = (nota1+nota2)/2
print(f"a media ponderada da suas duas notas é {med:.2f}")

digite sua nota:5.5
digite sua nota: 6.5
a media ponderada da suas duas notas é 6.00


In [ ]:
'''8-Escreva um programa que leia um valor em metros e o exiba convertido em centímetros e milímetros'''
valor = float(input('digite um valor em metros:'))
cent = valor*100
mili = valor * 1000
dec = valor /10
print(f' o valor digitado em metros, corresponde a {cent}cm \n {mili} milimetros \n e {dec} decametros')


digite um valor em metros:23.57
 o valor digitado em metros, corresponde a 2357.0cm 
 23570.0 milimetros 
 e 2.357 decametros


In [ ]:
'''9- Faça um programa que leia um número Inteiro qualquer e mostre na tela a sua tabuada.'''
num = int(input('digite um numero: '))
print('a tabuada desse numero de 1-10 corresponde a')
print(f' {num} x 1 = {num}\n {num} x 2 = {num*2} \n {num} x 3 = {num*3} \n {num} x 4 = {num*4} \n {num} x 5 = {num*5} \n {num} x 6 = {num*6} ')
print(f' {num} x 7 = {num*7} \n {num} x 8 = {num*8} \n {num} x 9 = {num*9} \n {num} x 10 = {num*10}')

digite um numero: 1500
a tabuada desse numero de 1-10 corresponde a
 1500 x 1 = 1500
 1500 x 2 = 3000 
 1500 x 3 = 4500 
 1500 x 4 = 6000 
 1500 x 5 = 7500 
 1500 x 6 = 9000 
 1500 x 7 = 10500 
 1500 x 8 = 12000 
 1500 x 9 = 13500 
 1500 x 10 = 15000


In [ ]:
'''10- Crie um programa que leia quanto dinheiro uma pessoa tem na carteira e mostre quantos dólares ela pode comprar.'''
din = float(input("digite quanto dinherio vc tem em reais: "))
dolar = din/5
print('o valor correspondente a isso em dolars é de ', dolar)

digite quanto dinherio vc tem em reais: 1500
o valor correspondente a isso em dolars é de  300.0


In [ ]:
'''11-Faça um programa que leia a largura e a altura de uma parede em metros, calcule a sua área e a quantidade de
tinta necessária para pintá-la, sabendo que cada litro de tinta pinta uma área de 2 metros quadrados.'''
lar = float(input("digite a largura da parade:"))
h = float(input("digite a altura: "))
area = h*lar
litros = area/2
print(f' sua parede tem {h}x{lar} e possui uma area de {area:.3f}m2')
print(f'para pinta-la vão ser nescessarios {litros:.3f} litros de tinta')


digite a largura da parade:3.08
digite a altura: 4.67
 sua parede tem 4.67x3.08 e possui uma area de 14.384m2
para pinta-la vão ser nescessarios 7.192 litros de tinta


In [ ]:
'''12-Faça um algoritmo que leia o preço de um produto e mostre seu novo preço, com 5% de desconto'''
prod = float(input("digite o valor do produto: "))
pre = prod -(prod*0.05)
print(f' o produto informado custava {prod}reais e agora com 5% de desconto, custa apenas {pre} reais')

digite o valor do produto: 120
 o produto informado custava 120.0reais e agora com 5% de desconto, custa apenas 114.0 reais


In [ ]:
'''13- Faça um algoritmo que leia o salário de um funcionário e mostre seu novo salário, com 15% de aumento.'''
sal = float(input('digite o seu salario:'))
new = sal+(sal*0.15)
print(f'o seu salario era {sal}reais e agora com 15% de aumento se tornara {new} reais' )

digite o seu salario:3500
o seu salario era 3500.0reais e agora com 15% de aumento se tornara 4025.0 reais


In [ ]:
'''14- Escreva um programa que converta uma temperatura digitando em graus Celsius e converta para graus Fahrenheit.'''
c = float(input("digite o grau celciun a ser convertido: "))
f = (9*c/5)+32
print(f' {c} graus celcius equivale a {f} fahrenheit')

digite o grau celciun a ser convertido: 24
 24.0 graus celcius equivale a 75.2 fahrenheit


In [ ]:
'''15-Escreva um programa que pergunte a quantidade de Km percorridos por um carro alugado e a quantidade de dias pelos quais ele foi alugado.
 Calcule o preço a pagar, sabendo que o carro custa R$60 por dia e R$0,15 por Km rodado.'''
km = float(input("digite a quantidade de kms rodados com o carro: "))
dias = int(input("digite a quantidade de dias que vc ficou com o carro: "))
preco = 60.0*dias+0.15*km
print(f'o preço custara {preco}reais ')

digite a quantidade de kms rodados com o carro: 4650
digite a quantidade de dias que vc ficou com o carro: 10
o preço custara 1297.5reais 


aula 08 - utilizando modulos

In [ ]:
import random
# random.shuffler(lista) mistura o itens de uma lita e cria uma nova lista
l =['maca','matheus','cereja','celul.ar']
random.shuffle(l)
print(l)

['cereja', 'celul.ar', 'maca', 'matheus']


In [ ]:
'''16- Crie um programa que leia um número Real qualquer pelo teclado e mostre na tela a sua porção Inteira.'''
import math
a= float(input("digite um numero: "))
b =float(input("digite um numero: "))
c = a/b
print(math.floor(c))
#math.floor seleciona e exibe apenas a porção inteira de um numero
print(f'{c:.3f}')

digite um numero: 14.56
digite um numero: 4.54
3
3.207


In [ ]:
'''17-Faça um programa que leia o comprimento do cateto oposto e do cateto adjacente de
 um triângulo retângulo. Calcule e mostre o comprimento da hipotenusa.'''
import math
catop=float(input("digite um numero: "))
catadj = float(input("digite um numero: "))
print(f'a hipotenusa é {math.hypot(catop,catadj):.3f}')



digite um numero: 12.4
digite um numero: 32.5
a hipotenusa é 34.785


In [ ]:
'''18 Faça um programa que leia um ângulo qualquer e mostre na tela o
valor do seno, cosseno e tangente desse ângulo.'''
import math
ang =float(input("digite o valor do angulo: "))
seno = math.sin(math.radians(ang))
cos = math.cos(math.radians(ang))
tang = math.tan(math.radians(ang))
print(f'o valor do cosseno: {cos:.2f}graus \n seno {seno:.2f} graus \n tangente:{tang:.2f} graus ')


digite o valor do angulo: 30
o valor do cosseno: 0.87graus 
 seno 0.50 graus 
 tangente:0.58 graus 


In [ ]:
'''19-Um professor quer sortear um dos seus quatro alunos para apagar o quadro. Faça um programa
que ajude ele, lendo o nome dos alunos e escrevendo na tela o nome do escolhido.'''
import random
lista=[]
a =str(input("digite o seu nome: "))
lista.append(a)
b =str(input("digite o seu nome: "))
lista.append(b)
c =str(input("digite o seu nome: "))
lista.append(c)
d =str(input("digite o seu nome: "))
lista.append(d)
print(f'dos alunos listadoos, o escolhido(a) foi {random.choice(lista)}')
#random.choice escolhe um item aleatoriamente de uma lista

digite o seu nome: a
digite o seu nome: d
digite o seu nome: f
digite o seu nome: x
dos alunos listadoos, o escolhido(a) foi f


In [ ]:
'''20-O mesmo professor do desafio 019 quer sortear a ordem de apresentação de trabalhos dos alunos.
Faça um programa que leia o nome dos quatro alunos e mostre a ordem sorteada.'''
import random
lista =[]
for c in range(4):
  nomes=str(input("digite o seu nome: "))
  lista.append(nomes)
  random.shuffle(lista) # entender melhor como funciona a função shuffle - ela mistura uma lista e gera uma nova lista
print(f' a ordem de apresentação dos trabalhos sera {lista}')
# a função shuffle embaralha um conjunto de valores e os oferece em uma nova ordem


digite o seu nome: matheus 
digite o seu nome: jair
digite o seu nome: miguel
digite o seu nome: vinicius
 a ordem de apresentação dos trabalhos sera ['matheus ', 'miguel', 'vinicius', 'jair']


In [ ]:
'''21-Faça um programa em Python que abra e reproduza o áudio de um arquivo MP3.'''
import pygame



In [ ]:
'''Exercício Python 022: Crie um programa que leia o nome completo de uma pessoa e mostre:
- O nome com todas as letras maiúsculas e minúsculas.
- Quantas letras ao todo (sem considerar espaços).
- Quantas letras tem o primeiro nome.'''
n =str(input("informe o seu nome completo: "))
print(f'seu nome com todas as letras em maiusculo é {n.upper()}') #frase.upper() coloca todos os caracteres em maiusculo
print(f'seu nome com todas as letras em minusculo é {n.lower()}') #frase.lower()reduz todos os caracteres da frase para o minusculo
print(f'a qtd de letras em seu nome é {len(n.strip())}')
a = n.split() # frase.split() trasnforma cada palavra existente na frase em um item de uma lista
print(f'o seu primeiro nome tem {len(a[0])}')#len(frase) calcula a qtd de caracteres que existe em uma frase


informe o seu nome completo: edilson
seu nome com todas as letras em maiusculo é EDILSON
seu nome com todas as letras em minusculo é edilson
a qtd de letras em seu nome é 7
o seu primeiro nome tem 7


In [ ]:
"""Exercício Python 023: Faça um programa que leia um número de 0 a 9999 e mostre na tela cada um dos dígitos separados"""
n = str(input("digite um numero inteiro entre 0 a 9999: "))
a = n.split()
if len(a[0]) == 4 :
  print(f'milhar: {a[0][0]} \n centena: {a[0][1]} \n dezena: {a[0][2]} \n unidade: {a[0][3]}')
if len(a[0])== 3:
  print(f'milhar:0 \n centena : {a[0][0]} \n  dezena:{a[0][1]} \n unidade: {a[0][2]}')
if len(a[0])==2:
  print(f' milhar: 0 \n centena: 0 \n dezena:{a[0][0]} \n unidade:{a[0][1]}')
if len(a[0])==1:
  print(f'milhar: 0 \n centena: 0 \n dezena: 0 \n unidade: {a[0][0]}')


digite um numero inteiro entre 0 a 9999: 123
milhar:0 
 centena : 1 
  dezena:2 
 unidade: 3


In [ ]:
''' exercicio do curso senac'''
nome  = str(input("digite seu nome: "))
idade =  int(input("digite sua idade : "))
falta  = 100 - idade
ano  = 2024 + falta
print(f' ola {nome} tudo bem? vc tem {idade} anos, voce tera 100 anos em {ano} ')

digite seu nome: mauy
digite sua idade : 17
 ola mauy tudo bem? vc tem 17 anos, voce tera 100 anos em 2107 


In [ ]:
'''24-Crie um programa que leia o nome de uma cidade diga se ela começa ou não com o nome "SANTO".'''
cida = str(input("digite o nome de uma cidade: "))
print('a cidade digitada foi ', cida.upper())
cidade = cida.upper()
a = cidade.split()
print(a)
if a[0] == 'SANTO' or a[0] == 'SANTA' or a[0]=='SÃO':
  print('a cidade digitada possui nome santo ')
else:
  print('a cidade não possui nome santo')

digite o nome de uma cidade: são paulo
a cidade digitada foi  SÃO PAULO
['SÃO', 'PAULO']
a cidade digitada possui nome santo 


In [ ]:
'''25-Crie um programa que leia o nome de uma pessoa e diga se ela tem "SILVA" no nome.'''
a =str(input("digite o seu nome: ")).upper()

if 'SILVA' in a: # o condando 'in'é usado para buscar uma palavra dentro da frase ex: 'palavra' in frase()
  print('o nome possui silva ')
else:
  print("a palavra silva não existe no nome")

digite o seu nome: matheus rocha beer
a palavra silva não existe no nome


In [ ]:
''' Faça um programa que leia uma frase pelo teclado e mostre quantas vezes aparece a letra "A",
em que posição ela aparece a primeira vez e em que posição ela aparece a última vez.'''
a =str(input("digite uma frase: "))
s =a.strip() # a função strip() retira os espaço desnecessarios da string
qtd_vezes = s.count('a')
primeira =s.find('a') #frase.find('caractere a ser econtrado') mostra o primeiro termo em que esta localizado o caractere
ultima = s.rfind('a') # montra a ultimo termo onde esta localizado o caractere
print(f' a frase digitada foi {s}')
print(f'''na frase digitada existe {qtd_vezes} vezes a letra 'a'\n ela aparece no termo {(primeira)+1} pela primeira
vez, no termo {(ultima)+1} pela ultima vez''')# é nescessario somar +1 no valor do termo pois o find busca começando pelo 0

digite uma frase:   casa
 a frase digitada foi casa
na frase digitada existe 2 vezes a letra 'a'
 ela aparece no termo 2 pela primeira 
vez, no termo 4 pela ultima vez


In [ ]:
'''27- Faça um programa que leia o nome completo de uma pessoa, mostrando em seguida o primeiro e o último nome separadamente.'''
nome =str(input("digite seu nome "))
a = nome.split()
print(f'seu primeiro nome é {a[0].upper()} e seu ultimo nome é {a[-1].upper()}')
print(a)

digite seu nome matheus rocha nemer 
seu primeiro nome é MATHEUS e seu ultimo nome é NEMER
['matheus', 'rocha', 'nemer']


28- Escreva um programa que faça o computador "pensar" em um número inteiro entre 0 e 5 e peça para o usuário tentar descobrir qual foi o número escolhido pelo computador. O programa deverá escrever na tela se o usuário venceu ou perdeu.

In [ ]:
import random
a = random.randint(0,5)
b = int(input("digite o numero que o computador pensou de 0 a 5: "))
if a==b:
  print('vc acertou')
else:
  print('vc errou ')

digite o numero que o computador pensou de 0 a 5: 4
vc acertou


29 -  Escreva um programa que leia a velocidade de um carro. Se ele ultrapassar 80Km/h, mostre uma mensagem dizendo que ele foi multado. A multa vai custar R$7,00 por cada Km acima do limite.

In [ ]:
v = float(input("digite a velocidade do carro: "))
if v>80.0:
   acima = v-80
   multa = 7*acima
   print(f'vc foi multado no valor de R$:{multa:.2f}')
else:
    print('vc não foi multado ')


digite a velocidade do carro: 85
vc foi multado no valor de R$:35.00


30 - Crie um programa que leia um número inteiro e mostre na tela se ele é PAR ou ÍMPAR

In [ ]:
a =  int(input("digite um numero: "))
if a%2==0:
  print('é um numero par')
else:
  print('é um numero impar')

digite um numero: 5678
é um numero par


*31: Desenvolva um programa que pergunte a distância de uma viagem em Km. Calcule o preço da passagem, cobrando R$0,50  por Km para viagens de até 200Km e R$0,45 parta viagens mais longas.*

In [ ]:
d = float(input("digite a distancia de sua viagem "))
valor =  d*0.5 if d<=200 else d*0.45
print(f'o valor a ser pago pela sua viagem sera de R$:{valor}')

digite a distancia de sua viagem 201
o valor a ser pago pela sua viagem sera de R$:90.45


Exercício Python 032: Faça um programa que leia um ano qualquer e mostre se ele é bissexto.

In [ ]:
ano = int(input('digite um ano:'))
if ano%4==0 and ano%100!=0 or ano%400==0:
  print('ano é bissexto')
else:
  print('não é bissexto')

digite um ano:2016
ano é bissexto


033: Faça um programa que leia três números e mostre qual é o maior e qual é o menor.

In [ ]:
menor = 0
a = int(input("digite um numero: "))
b = int(input("digite um numero: "))
c = int(input("digite um numero: "))
menor = a
maior = b
if menor<b and menor <c:
  menor = a
elif menor >b:
   menor = b
   #print(f'o menor numero é: {menor}')
if menor >c:
    menor = c
print(f'o menor numero é: {menor}')
if  maior >a and maior >c:
  maior = b
elif a>maior and a>c:
  maior=a
elif c>maior and c>a:
  maior = c
print(f'o maior numero é {maior}')



digite um numero: 1
digite um numero: 2
digite um numero: 3
o menor numero é: 1
o maior numero é 3


34: Escreva um programa que pergunte o salário de um funcionário e calcule o valor do seu aumento. Para salários superiores a R$1250,00, calcule um aumento de 10%. Para os inferiores ou iguais, o aumento é de 15%.

In [ ]:
sal = float(input("digite seu salario: "))
if sal>1250:
  aumento = sal+(sal*0.1)
else:
  aumento = sal+(sal*0.15)
print(f"seu salario novo sera de {aumento}")

digite seu salario: 1250
seu salario novo sera de 1437.5


 35 - Desenvolva um programa que leia o comprimento de três retas e diga ao usuário se elas podem ou não formar um triângulo.

In [ ]:
a = float(input("digite o lado 1 do traingulo: "))
b = float(input("digite o lado 2 do triangulo: "))
c= float(input("digite o lado 3 do triangulo: "))
if (b-c<a<b+c) and (a-c<b<a+c) and (a-b<c<b+a):
  print("as retas formam um triangulo!")
else:
  print("as  retas não formam um triangulo. ")

digite o lado 1 do traingulo: 2
digite o lado 2 do triangulo: 3
digite o lado 3 do triangulo: 5
as  retas não formam um triangulo. 


036: Escreva um programa para aprovar o empréstimo bancário para a compra de uma casa. Pergunte o valor da casa, o salário do comprador e em quantos anos ele vai pagar. A prestação mensal não pode exceder 30% do salário ou então o empréstimo será negado.

In [ ]:
sal = float(input("digite o valor do seu salario: "))
anos = int(input('digite em quantos anos vc pretende pagar a sua casa: '))
casa = float(input("digite o valor da casa: "))
parcela = casa/(anos*12)
parcela_max = sal*0.3
if parcela <= parcela_max:
  print(f'emprestimo aprovado, o valor de sua parcela sera de {parcela:.2f} reais e vc pode pagar em {anos} anos ')
elif parcela>parcela_max:
  print(f' emprestimo negado, o valor da parcela esta acima de 30% do seu salario. ')

digite o valor do seu salario: 1500
digite em quantos anos vc pretende pagar a sua casa: 30
digite o valor da casa: 100000
emprestimo aprovado, o valor de sua parcela sera de 277.78 reais e vc pode pagar em 30 anos 


37: Escreva um programa em Python que leia um número inteiro qualquer e peça para o usuário escolher qual será a base de conversão: 1 para binário, 2 para octal e 3 para hexadecimal.

In [ ]:
n = int(input("digite um numero: "))
print('=+'*20)
conv = int(input("digite 1- conversão para binario 2- conversão para octal 3 - conversão para hexadeximal "))
if conv ==1:
  print(f'O numero em binario é {bin(n)[2:]}')
elif conv ==2:
  print(f'o numero {n} em sistema octal é {oct(n)[2:]}')
elif conv ==3:
  print(f'o numero {n} convertido em hexadeximal é {hex(n)[2:]}')

digite um numero: 1535
=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+=+
digite 1- conversão para binario 2- conversão para octal 3 - conversão para hexadeximal 2
o numero 1535 em sistema octal é 2777


38: Escreva um programa que leia dois números inteiros e compare-os. mostrando na tela uma mensagem:
- O primeiro valor é maior
- O segundo valor é maior
- Não existe valor maior, os dois são iguais

In [ ]:
a = int(input("digite um numero: "))
b = int(input("digite um numero: "))
if a>b:
  print('o primeiro numero e´maior ')
elif b>a:
  print('o segundo numero é maior')
else:
  print('os dois são iguais')

digite um numero: 12
digite um numero: 12
os dois são iguais


Faça um programa que leia o ano de nascimento de um jovem e informe, de acordo com a sua idade, se ele ainda vai se alistar ao serviço militar, se é a hora exata de se alistar ou se já passou do tempo do alistamento. Seu programa também deverá mostrar o tempo que falta ou que passou do prazo.



In [ ]:
ano = int(input("digite seu ano de nascimento: "))
atual = 2024
idade = atual - ano
if idade ==18:
  print("é hora de se alistar")
elif idade < 18:
  print(f'ainda não é hora de se alistar, vc deve se alistar daqui {18-idade} anos em {2024+(18-idade)}')
elif idade>18:
  print(f' vc esta atrasado,  vc devia ter se alistado a {idade-18} anos atras, va a uma junta militar e se aliste o mais rapido possivel. ')

digite seu ano de nascimento: 2016
ainda não é hora de se alistar, vc deve se alistar daqui 10 anos em 2034


40: Crie um programa que leia duas notas de um aluno e calcule sua média, mostrando uma mensagem no final, de acordo com a média atingida:
- Média abaixo de 5.0: REPROVADO
- Média entre 5.0 e 6.9: RECUPERAÇÃO
- Média 7.0 ou superior: APROVADO

In [ ]:
n1 = float(input("digite sua nota: "))
n2 = float(input("digite sua nota: "))
med = (n1+n2)/2
if med<5:
  print('reprovado')
elif med >=5.0 and med<=6.9:
  print("vc esta de recuperação")
elif med >=7:
  print("vc esta aprovado.")

digite sua nota: 6
digite sua nota: 7
vc esta de recuperação


41: A Confederação Nacional de Natação precisa de um programa que leia o ano de nascimento de um atleta e mostre sua categoria, de acordo com a idade:
- Até 9 anos: MIRIM
- Até 14 anos: INFANTIL
- Até 19 anos: JÚNIOR
- Até 25 anos: SÊNIOR
- Acima de 25 anos: MASTER

In [ ]:
ano = int(input("digite o ano de seu nascimento: "))
idade = 2024 - ano
if idade <=9:
  print("categoria mirim")
if 9<idade<=14:
  print("categoria infantil")
elif 14<idade <=19:
  print('categoria junior')
elif 19<idade<=25:
  print("categoria senior")
elif idade>25:
  print("categoria master")

digite o ano de seu nascimento: 2000
categoria senior


42: Refaça o DESAFIO 035 dos triângulos, acrescentando o recurso de mostrar que tipo de triângulo será formado:
- EQUILÁTERO: todos os lados iguais
- ISÓSCELES: dois lados iguais, um diferente
- ESCALENO: todos os lados diferentes

In [ ]:
l1 = float(input("digite o valor do lado1 do triangulo: "))
l2 = float(input("digite o valor do lado2 do triangulo: "))
l3 = float(input("digite o valor do lado3 do triangulo: "))
if (l1<l2+l3) and (l2<l1+l3) and (l3<l2+l1):
  print('os lados atendem as condições de existencia e o triangulo pode existir')
  if l1==l2==l3:
    print('o triangulo é equilatero')
  if (l1==l2 and l3!=l1 and l3!=l2) or (l2==l3 and l1!=l2 and l1!=l3) or (l3==l1 and l2!=l3 and l2!=l1):
    print('o triangulo é isosceles')
  elif l1!=l2!=l3:
    print("o triangulo é escaleno")
else:
  print("os lados não atingem as condições de existencia dos triangulos")


digite o valor do lado1 do triangulo: 2
digite o valor do lado2 do triangulo: 3
digite o valor do lado3 do triangulo: 4
os lados atendem as condições de existencia e o triangulo pode existir
o triangulo é escaleno


43: Desenvolva uma lógica que leia o peso e a altura de uma pessoa, calcule seu Índice de Massa Corporal (IMC) e mostre seu status, de acordo com a tabela abaixo:
- IMC abaixo de 18,5: Abaixo do Peso
- Entre 18,5 e 25: Peso Ideal
- 25 até 30: Sobrepeso
- 30 até 40: Obesidade
- Acima de 40: Obesidade Mórbida

In [ ]:
kg = float(input("digite o seu peso corporal"))
h = float(input("digite sua altura: "))
imc = kg/(h**2)
if imc<18.5:
  print("abaixo do peso")
elif imc>=18.5 and imc<=25:
  print("peso ideal")
elif 25<imc<=30:
  print("sobrepeso")
elif 30<imc<=40:
  print('obesidade')
elif imc>40:
  print("obesidade morbida")

digite o seu peso corporal81
digite sua altura: 1.81
peso ideal


44: Elabore um programa que calcule o valor a ser pago por um produto, considerando o seu preço normal e condição de pagamento:
- à vista dinheiro/cheque: 10% de desconto
- à vista no cartão: 5% de desconto
- em até 2x no cartão: preço formal
- 3x ou mais no cartão: 20% de juros

In [ ]:
print('''bem vindo ao nosso sistema de pagamentos escolha a opção de pagamento abaixo:
1 - dinheiro
2 - debito
3 - 2x cartão
4 - 3x ou mais no cartão ''')
pa = int(input("digite a opção de pagamento: "))
preco  = float(input("digite o valor da compra: "))
if pa ==1:
  v = preco - (preco*0.1)
  print(f'O valor a ser pago sera de {v} reais')
elif pa==2:
  v = preco - (preco*0.05)
  print(f"valor a ser pago sera de {v}")
elif pa ==3:
  v = preco
  print(f"O valor a ser pago sera de {v}")
elif pa==4:
  v = preco +(preco*0.2)
  print(f'O valor a ser pago é de {v}')
else:
  print('opção de pagamento invalida')

bem vindo ao nosso sistema de pagamentos escolha a opção de pagamento abaixo: 
1 - dinheiro 
2 - debito 
3 - 2x cartão 
4 - 3x ou mais no cartão 
digite a opção de pagamento: 5
digite o valor da compra: 12
opção de pagamento invalida


45: Crie um programa que faça o computador jogar Jokenpô com você.

In [ ]:
import random
print('''vamos jogar jokempo
1-  pedra
2 - papel
3 - tesoura ''')
jogada = int(input("digite sua jogada"))
pc = random.randint(1,3)
if jogada==1 and pc ==1:
  print('empate')
elif jogada==1 and pc==2:
  print(" eu venci")
elif jogada==1 and pc==3:
  print("vc venceu")
elif jogada==2 and pc==2:
  print('empate')
elif jogada==2 and pc ==1:
  print("vc venceu")
elif jogada==2 and pc ==3:
  print('pc venceu')
elif jogada==3 and pc==3:
  print('empate')
elif jogada==3 and pc ==2:
  print('pc venceu ')
elif jogada ==3 and pc ==1:
  print("vc venceu ")
else:
  print('opção invalida')

vamos jogar jokempo
1-  pedra 
2 - papel 
3 - tesoura 
digite sua jogada7
opção invalida


46: Faça um programa que mostre na tela uma contagem regressiva para o estouro de fogos de artifício, indo de 10 até 0, com uma pausa de 1 segundo entre eles.

In [ ]:
import time
print('CONTAGEM REGRESSIVA!!!!')
for i in range(10,-1,-1):
  print(i)
  time.sleep(1)
print('BOOOOM')

CONTAGEM REGRESSIVA!!!!
10
9
8
7
6
5
4
3
2
1
0
BOOOOM


47: Crie um programa que mostre na tela todos os números pares que estão no intervalo entre 1 e 50.

In [ ]:
cont = 0
for i in range(1,51):
  if i%2==0:
    cont+=1
    print(i)
print(f'a qtd de numeros pares entre 1 e 50 é {cont}')


2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
a qtd de numeros pares entre 1 e 50 é 25


48: Faça um programa que calcule a soma entre todos os números que são múltiplos de três e que se encontram no intervalo de 1 até 500.


In [ ]:
soma =  0
for c in range(1,501):
  if c%3==0:
    soma +=c
print(f'a soma de todos os multiplos de 3 no intervalo de 1 a 500 é {soma}')

a soma de todos os multiplos de 3 no intervalo de 1 a 500 é 41583


49: Refaça o DESAFIO 009, mostrando a tabuada de um número que o usuário escolher, só que agora utilizando um laço for.

In [ ]:
n = int(input("digite um numero: "))
for c in range(1,11):
  print(f'{n}x{c}={n*c}')


digite um numero: 8
8x1=8
8x2=16
8x3=24
8x4=32
8x5=40
8x6=48
8x7=56
8x8=64
8x9=72
8x10=80


50: Desenvolva um programa que leia seis números inteiros e mostre a soma apenas daqueles que forem pares. Se o valor digitado for ímpar, desconsidere-o.

In [ ]:
soma = 0
for i in range(1,7):
  n = int(input("digite um numero: "))
  if n%2==0:
    soma+=n
print(f'a soma dos numeros pares é {soma}')

digite um numero: 1
digite um numero: 2
digite um numero: 3
digite um numero: 4
digite um numero: 5
digite um numero: 6
a soma dos numeros pares é 12


51: Desenvolva um programa que leia o primeiro termo e a razão de uma PA. No final, mostre os 10 primeiros termos dessa progressão.

In [ ]:
n = int(input("digite o primeiro termo de uma pa: "))
r = int(input("digite a razão da pa: "))
for i in range(1,11):
  n +=r
  print(n)

digite o primeiro termo de uma pa: 2
digite a razão da pa: 1
3
4
5
6
7
8
9
10
11
12


52: Faça um programa que leia um número inteiro e diga se ele é ou não um número primo.

In [ ]:
c=0
n = int(input("digite um numero: "))
for i in range(1,n+1):
  if n%i==0:
    c+=1
if c>2:
    print(f'o numero não é primo pois ele foi divido por {c} numeros ')
elif c==2:
    print("o numero é primo")


digite um numero: 543213
o numero não é primo pois ele foi divido por 32 numeros 


53: Crie um programa que leia uma frase qualquer e diga se ela é um palíndromo, desconsiderando os espaços.

Exemplos de palíndromos: APOS A SOPA, A SACADA DA CASA, A TORRE DA DERROTA, O LOBO AMA O BOLO, ANOTARAM A DATA DA MARATONA.

In [ ]:
frase = str(input("digite uma frase: "))
a = frase.replace(' ','')
t = (a[::-1])
if a==t:
  print("a frase é um palindromo")
else:
  print("a frase não é um palindromo")

digite uma frase: a torre da derrota apos a sopa a sacada da casa o lobo ama o bolo anotaram a data da maratona 
a frase não é um palindromo


54: Crie um programa que leia o ano de nascimento de sete pessoas. No final, mostre quantas pessoas ainda não atingiram a maioridade e quantas já são maiores.


In [ ]:
maior = 0
menor = 0
for i in range(1,8):
  n = int(input("digite o ano de nascimento: "))
  idade =  2024 - n
  if idade >=18:
    maior +=1
  elif idade<18:
    menor+=1
print(f'maiores de idade: {maior} pessoas \n menores de idade: {menor} pessoas')


digite o ano de nascimento: 2019
digite o ano de nascimento: 2020
digite o ano de nascimento: 2021
digite o ano de nascimento: 2023
digite o ano de nascimento: 2000
digite o ano de nascimento: 2001
digite o ano de nascimento: 2017
maiores de idade: 2 pessoas 
 menores de idade: 5 pessoas


55) Faça um programa que leia o peso de cinco pessoas. No final, mostre qual foi o maior e o menor peso lidos.

In [ ]:
menor = 1000
maior  =  0
for i in range(1,6):
   n = float(input("digite o seu peso em kg: "))
   if n>maior:
    maior=n
   if n<menor:
    menor = n
print(f' o maior peso lido é: {maior} kgs \n menor peso: {menor} kgs')

digite o seu peso em kg: 1
digite o seu peso em kg: 2
digite o seu peso em kg: 3
digite o seu peso em kg: 4
digite o seu peso em kg: 5
 o maior peso lido é: 5.0 kgs 
 menor peso: 1.0 kgs


56: Desenvolva um programa que leia o nome, idade e sexo de 4 pessoas. No final do programa, mostre: a média de idade do grupo, qual é o nome do homem mais velho e quantas mulheres têm menos de 20 anos.

In [ ]:
h = 0
m=0
med=0
velho = 0
menor = 0
for i in range(4):
  idade=  int(input("digite sua idade"))
  sexo = int(input("digite seu sexo 1- masculino 2- feminino 3- outro"))
  nome = str(input('digite seu nome:'))
  med +=idade
  if sexo==1:
    h+=1
    if idade>velho:
      velho = idade
      nome_velho = nome
  if sexo ==2:
    m+=1
    if idade <20:
      menor+=1
print(f' o grupo é fromado por {h} homens e {m} mulheres, o homem mais velho se chama {nome_velho} \n existem {menor} mulheres com menos de 20 anos no grupo e a media das idades é {(med/4)} anos')

digite sua idade24
digite seu sexo 1- masculino 2- feminino 3- outro1
digite seu nome:matheus
digite sua idade28
digite seu sexo 1- masculino 2- feminino 3- outro1
digite seu nome:lucas
digite sua idade56
digite seu sexo 1- masculino 2- feminino 3- outro2
digite seu nome:marcela
digite sua idade34
digite seu sexo 1- masculino 2- feminino 3- outro2
digite seu nome:julia
 o grupo é fromado por 2 homens e 2 mulheres, o homem mais velho se chama lucas 
 existem 0 mulheres com menos de 20 anos no grupo e a media das idades é 35.5 anos


 57: Faça um programa que leia o sexo de uma pessoa, mas só aceite os valores 'M' ou 'F'. Caso esteja errado, peça a digitação novamente até ter um valor correto.

In [ ]:
while True:
  s = str(input("digite seu sexo: M- masculino F-feminino")).upper()
  if s=='M':
    print("homem")
  elif s=='F':
    print("mulher")
  else:
    print("sexo invalido")
    break

digite seu sexo: M- masculino F-femininom
homem
digite seu sexo: M- masculino F-femininof
mulher
digite seu sexo: M- masculino F-femininos
sexo invalido


58: Melhore o jogo do DESAFIO 028 onde o computador vai "pensar" em um número entre 0 e 10. Só que agora o jogador vai tentar adivinhar até acertar, mostrando no final quantos palpites foram necessários para vencer.


In [ ]:
import random
print("Sou o seu computador, acabei de pensar em um numero entre 0 e 10")
p = random.randint(0,10)
c =0
while n!=p:
  n =int(input("digite o numero que o computador pensou entre 0 e 10: "))
  c+=1
  if n==p:
    break
print(f'vc levou {c} tentativas para acertar o numero que o computador pensou, o numero por ele pensado é {p}')

Sou o seu computador, acabei de pensar em um numero entre 0 e 10
digite o numero que o computador pensou entre 0 e 10: 4
digite o numero que o computador pensou entre 0 e 10: 6
digite o numero que o computador pensou entre 0 e 10: 8
digite o numero que o computador pensou entre 0 e 10: 2
digite o numero que o computador pensou entre 0 e 10: 1
digite o numero que o computador pensou entre 0 e 10: 3
digite o numero que o computador pensou entre 0 e 10: 10
digite o numero que o computador pensou entre 0 e 10: 0
digite o numero que o computador pensou entre 0 e 10: 9
digite o numero que o computador pensou entre 0 e 10: 7
vc levou 10 tentativas para acertar o numero que o computador pensou, o numero por ele pensado é 7


59: Crie um programa que leia dois valores e mostre um menu na tela:
[ 1 ] somar
[ 2 ] multiplicar
[ 3 ] maior
[ 4 ] novos números
[ 5 ] sair do programa
Seu programa deverá realizar a operação solicitada em cada caso.

In [ ]:
numeros = []
c= 0
mult = 1
while True:
  n1 = int(input("digite um nuemro: "))
  n2 = int(input("digite um  numero: "))
  menu = int(input('''O QUE VC QUER FAZER?

  [1] SOMAR
  [2] MULTIPLICAR
  [3] MAIOR
  [4] NOVOS NUMEROS
  [5] SAIR '''))
  numeros.append(n1)
  numeros.append(n2)
  if menu==1:
    c = n1+n2

    print(f'soma={n1}+{n2}={c}')

  elif menu==2:
    mult = n1*n2

    print(f'multiplicação: {n1}x{n2}={(mult)}')
  elif menu==3:

      print(f'o maior numero é {max(numeros)}')


  elif menu==4:
    n = int(input("digite um numero: "))
    menu = int(input('''O QUE VC QUER FAZER?

  [1] SOMAR
  [2] MULTIPLICAR
  [3] MAIOR
  [4] NOVOS NUMEROS
  [5] SAIR '''))
    numeros.append(n)
    if menu==1:
      for i in numeros:
        c+=i
      print(f'soma = {n1}+{n2}+{n}={c}')
    elif menu==2:
        for i in numeros:
          mult*=i
        print(f'multiplicção ={n1}x{n2}x{n}= {mult}')
    elif menu==3:
      print(f'o maior numero é {max(numeros)}')


  elif menu==5:
    print('programa encerrado')
    break
  elif menu>5:
    print('opção invalida')





o maior numero é3
multiplicção =2x4x8=3072


60: Faça um programa que leia um número qualquer e mostre o seu fatorial.

Ex: 5! = 5 x 4 x 3 x 2 x 1 = 120

In [ ]:
n = int(input('digite um numero: '))
c =1
for i in range(1,n+1):
  c*=i
print(f'o fatorial de {n} é {c}')

digite um numero: 6
o fatorial de 6 é 720


usando while

In [ ]:
n = int(input('digite um numero: '))
c =1
soma = 1
somaf=1
while c<=n:
  soma =c*1 # soma sempre sera atualizada para um novo numero pois c sera atualizado em +1 ate que c seja igual a n
  c+=1
  somaf *=soma  # soma fimal sempre sera multiplicada pelo novo valor de soma ate que c==10
print(f'o fatorial de {n}: {somaf}')


digite um numero: 5
o fatorial de 5: 120


61: Refaça o DESAFIO 051, lendo o primeiro termo e a razão de uma PA, mostrando os 10 primeiros termos da progressão usando a estrutura while.

In [ ]:
c= 0
n = int(input("digite o primeiro termo da pa:"))
r = int(input("digite a razão da pa: "))

while True:
  if c<=10:
    n+=r
    print(f'termo{c}={n}')
    c+=1
  else:
    break


digite o primeiro termo da pa:2
digite a razão da pa: 2
termo0=4
termo1=6
termo2=8
termo3=10
termo4=12
termo5=14
termo6=16
termo7=18
termo8=20
termo9=22
termo10=24


62- perguntando para o usuário se ele quer mostrar mais alguns termos. O programa encerrará quando ele disser que quer mostrar 0 termos.

In [ ]:
c= 0
n = int(input("digite o primeiro termo da pa: "))
r = int(input("digite a razão da pa: "))
while True:
  if c<=10:
    n+=r
    c+=1
    print(f'termo{c}={n}')
  if c==10:
    while True:

      op =int(input("digite quantos termos a mais vc deseja ver desta PA:"))
      if op!=0:
        for i in range(c,(c+op)+1):
          n+=r
          print(f'termo{i}={n}')
          c+=1
      if op==0:
        print("encerrado")
        break


termo1=5
termo2=7
termo3=9
termo4=11
termo5=13
termo6=15
termo7=17
termo8=19
termo9=21
termo10=23
termo10=25
termo11=27
termo12=29
termo13=31
termo14=33
termo15=35
termo16=37
termo17=39
termo18=41
termo19=43
termo20=45
termo21=47
termo22=49
termo23=51
termo24=53
encerrado


KeyboardInterrupt: 

64- Escreva um programa que leia um número N inteiro qualquer e mostre na tela os N primeiros elementos de uma Sequência de Fibonacci.

Ex: 0 - 1 - 1 - 2 - 3 - 5 - 8

In [ ]:

n = int(input("digite quantos termos v quer mostrar: "))
t1 =0
t2 =1

print(t1,t2, end=' - ')
for i in range(3,n+1):
  t3=t1+t2
  print(t3,end='-')
  t1=t2
  t2=t3
print('fim')


digite quantos termos v quer mostrar: 23
0 1 - 1-2-3-5-8-13-21-34-55-89-144-233-377-610-987-1597-2584-4181-6765-10946-17711-fim


64: Crie um programa que leia vários números inteiros pelo teclado. O programa só vai parar quando o usuário digitar o valor 999, que é a condição de parada. No final, mostre quantos números foram digitados e qual foi a soma entre eles (desconsiderando o flag).

In [ ]:

c,soma,n =0,0,0
n = int(input("digite um numero: "))
while n!=999:
  c+=1
  soma+=n
  n = int(input("digite um numero: "))
print(f'VC digitou {c} numeros, e a soma dos numeros digitados e´de {soma}')
print("encerrado")

digite um numero: 2
digite um numero: 2
digite um numero: 2
digite um numero: 2
digite um numero: 999
VC digitou 4 numeros, e a soma dos numeros digitados e´de 8
encerrado


65: Crie um programa que leia vários números inteiros pelo teclado. No final da execução, mostre a média entre todos os valores e qual foi o maior e o menor valores lidos. O programa deve perguntar ao usuário se ele quer ou não continuar a digitar valores.

In [ ]:
op = 0
c=0
soma=0
maior, menor =0,0
l = []
while True:
  n = int(input("digite um numero: "))
  l.append(n)
  c+=1
  soma+=n
  op = int(input("digite 1 para continuar e 0 para sair: "))
  if op==0:
    break
print(f'vc digitou  {c} numeros, a media deles é {soma/c}')
print(f'o maior numero é {max(l)}')
print(f'o menor valor é {min(l)}')


digite um numero: 12
digite 1 para continuar e 0 para sair: 1
digite um numero: 11
digite 1 para continuar e 0 para sair: 0
vc digitou  2 numeros, a media deles é 11.5
o maior numero é 12
o menor valor pe 11


 66 - Crie um programa que leia números inteiros pelo teclado. O programa só vai parar quando o usuário digitar o valor 999, que é a condição de parada. No final, mostre quantos números foram digitados e qual foi a soma entre elas (desconsiderando o flag).


In [ ]:
c,soma=0,0
while True:
  n = int(input("digite um  numero: "))
  if n==999:
    c+=1
    print(f' vc precisou digitar {c-1} numeros ate digitar a condição de parada. a soma destes numeros é {soma} ')
    break
  if n!=999:
    c+=1
    soma+=n


digite um  numero: 1
digite um  numero: 2
digite um  numero: 3
digite um  numero: 4
digite um  numero: 5
digite um  numero: 999
 vc precisou digitar 5 numeros ate digitar a condição de parada. a soma destes numeros é 15 


67: Faça um programa que mostre a tabuada de vários números, um de cada vez, para cada valor digitado pelo usuário. O programa será interrompido quando o número solicitado for negativo.

In [ ]:
n=0
c=0
soma=0
while True:
  n= int(input("digite um numero: "))
  if n>=0:
    c+=1
    soma+=n
    for i in range(1,11):
      print(f'{n}x{i}={n*i}')

  if n<0:
    break
print(f' vc digitou {c} numeros ate digitar um numero negativo. a soma destes numeros é {soma}')
print("saindo...")
# basicamente esse codigo estabelece uma condição de parada quando o usuario digitar um  numero negativo
#o while true estabelece que enquanto ele for verdadeiro ele executara o comando indefinidamente.
# a unica maneira de parar o laço while true é criar uma condição dentro dele com o comando break


digite um numero: -2
 vc digitou 0 numeros ate digitar um numero negativo. a soma destes numeros é 0
saindo...


68: Faça um programa que jogue par ou ímpar com o computador. O jogo só será interrompido quando o jogador perder, mostrando o total de vitórias consecutivas que ele conquistou no final do jogo.

In [ ]:
j,computador=0,0
import random
soma,c,v =0,0,0
while True:
  n = int(input("digite um numero: "))
  a = random.randint(0,100)
  soma=n+a
  if soma%2==0:
    v+=1
    c+=1
  else:
    c+=1
    break
print(f'vc jogou {c} e venceu {v} vezes')




digite um numero: 2
digite um numero: 4
digite um numero: 3
digite um numero: 1
digite um numero: 1
digite um numero: 1
vc jogou 6 e venceu 5 vezes


69: Crie um programa que leia a idade e o sexo de várias pessoas. A cada pessoa cadastrada, o programa deverá perguntar se o usuário quer ou não continuar. No final, mostre:
A) quantas pessoas tem mais de 18 anos.
B) quantos homens foram cadastrados.
C) quantas mulheres tem menos de 20 anos.

In [ ]:
maioridade,homens,mulheres20=0,0,0
while True:
  idade = int(input("digite sua idade: "))
  sexo = int(input("digite seu sexo (1)- homem  (2)- mulher: "))

  if idade>18:
    maioridade+=1
  if sexo==1:
    homens+=1
  if sexo==2:
    if idade<20:
      mulheres20+=1
  c = int(input("deseja continuar: (1)sim (2)não"))
  if c==2:
    break
print(f'pessoas maiores de 18: {maioridade}')
print(f'homens: {homens}')
print(f'mulheres menores de 20: {mulheres20}')

digite sua idade: 23
digite seu sexo (1)- homem  (2)- mulher: 1
deseja continuar: (1)sim (2)não1
digite sua idade: 24
digite seu sexo (1)- homem  (2)- mulher: 2
deseja continuar: (1)sim (2)não1
digite sua idade: 12
digite seu sexo (1)- homem  (2)- mulher: 2
deseja continuar: (1)sim (2)não1
digite sua idade: 16
digite seu sexo (1)- homem  (2)- mulher: 2
deseja continuar: (1)sim (2)não1
digite sua idade: 24
digite seu sexo (1)- homem  (2)- mulher: 1
deseja continuar: (1)sim (2)não1
digite sua idade: 19
digite seu sexo (1)- homem  (2)- mulher: 2
deseja continuar: (1)sim (2)não2
pessoas maiores de 18: 4
homens: 2
mulheres menores de 20: 3


70: Crie um programa que leia o nome e o preço de vários produtos. O programa deverá perguntar se o usuário vai continuar ou não. No final, mostre:
A) qual é o total gasto na compra.
B) quantos produtos custam mais de R$1000.
C) qual é o nome do produto mais barato.

In [ ]:
s=0
while True:
  nome =str(input("digite o  numero do produto"))
  preco = float(input("digite o preço do produto: "))
  c =int(input("deseja continuar: (1)sim (2)não"))
  if c==2:
    break


71: Crie um programa que simule o funcionamento de um caixa eletrônico. No início, pergunte ao usuário qual será o valor a ser sacado (número inteiro) e o programa vai informar quantas cédulas de cada valor serão entregues.
OBS: considere que o caixa possui cédulas de R$50, R$20, R$10 e R$1.

In [ ]:
n20 = 0
n10 = 0
n1 = 0
n50 = 0
va = int(input("digite o valor que vc quer sacar: "))
new =0
new =va
while va>1:
  if va>50 or va==50:
    va = va -50
    n50+=1
  elif va>=20:
    va-=20
    n20+=1
  elif va>=10:
    va-=10
    n10+=1
  elif va>=1:
    va-=1
    n1+=1
  elif va<1:
    break
print(f'valor sacado: {new} \n notas de 50: {n50} \n nota de 20: {n20}\n nota de 10: {n10} \n nota de 1:{n1+1}')




digite o valor que vc quer sacar: 1327
valor sacado: 1327 
 notas de 50: 26 
 nota de 20: 1
 nota de 10: 0 
 nota de 1:7


# TUPLAS:
coleções de dados compostas e imutaveis


72 - Crie um programa que tenha uma dupla totalmente preenchida com uma contagem por extenso, de zero até vinte. Seu programa deverá ler um número pelo teclado (entre 0 e 20) e mostrá-lo por extenso.

In [ ]:
n = 'zero','um','dois','tres','quatro','cinco','seis','sete','oito','nove','dez','onze','doze','treze','quatorze','quinze','dezesseis','dezessete','dezoito','dezenove','vinte'
while True:
  num = int(input("digite um numero entre 0 a 20: "))
  if 0<=num<=20:
    print(f'vc digitou o numero {n[num]}')
    break
  if num  not in range(0,21):
    print('numero invalido')


digite um numero entre 0 a 20: 122
numero invalido
digite um numero entre 0 a 20: 211
numero invalido
digite um numero entre 0 a 20: 1
vc digitou o numero um


73: Crie uma tupla preenchida com os 20 primeiros colocados da Tabela do Campeonato Brasileiro de Futebol, na ordem de colocação. Depois mostre:
a) Os 5 primeiros times.
b) Os últimos 4 colocados.
c) Times em ordem alfabética.
d) Em que posição está o time da Chapecoense.

In [ ]:
times_brasileirao_2024 = (
    "Athletico-PR", "Flamengo", "Botafogo", "São Paulo", "Bahia",
    "Cruzeiro", "Atlético-MG", "Red Bull Bragantino", "Palmeiras",
    "Internacional", "Fortaleza", "Vasco da Gama",'chapecoence', "Grêmio",
    "Atlético-GO", "Cuiabá", "Corinthians", "Fluminense", "Vitória",
    "Santos", "Goiás"
)
print(f'os 5 primeiros times são:{times_brasileirao_2024[0:5]}')
print(f'os 4 ultimos times são: {times_brasileirao_2024[-4:]}')
print(f'times em ordem alfabética: {sorted(times_brasileirao_2024)}')
a = times_brasileirao_2024.index('chapecoence')
print(f'chapecoence esta na {a} posição.')

os 5 primeiros times são:('Athletico-PR', 'Flamengo', 'Botafogo', 'São Paulo', 'Bahia')
os 4 ultimos times são: ('Fluminense', 'Vitória', 'Santos', 'Goiás')
times em ordem alfabética: ['Athletico-PR', 'Atlético-GO', 'Atlético-MG', 'Bahia', 'Botafogo', 'Corinthians', 'Cruzeiro', 'Cuiabá', 'Flamengo', 'Fluminense', 'Fortaleza', 'Goiás', 'Grêmio', 'Internacional', 'Palmeiras', 'Red Bull Bragantino', 'Santos', 'São Paulo', 'Vasco da Gama', 'Vitória', 'chapecoence']
chapecoence esta na 12 posição.


74: Crie um programa que vai gerar cinco números aleatórios e colocar em uma tupla. Depois disso, mostre a listagem de números gerados e também indique o menor e o maior valor que estão na tupla.

In [ ]:
import random
tupla= tuple(random.randint(0,29) for i in range(10))
print(f' a tupla gerada por numeros aleatorios é  {tupla}')
print(f'o menor valor é {min(tupla)}')
print(f'o maior valor é {max(tupla)}')


 a tupla gerada por numeros aleatorios é  (9, 29, 26, 6, 2, 13, 25, 0, 15, 20)
o menor valor é 0
o maior valor é 29


75: Desenvolva um programa que leia quatro valores pelo teclado e guarde-os em uma tupla. No final, mostre:

A) Quantas vezes apareceu o valor 9.
B) Em que posição foi digitado o primeiro valor 3.
C) Quais foram os números pares

In [ ]:
a =int(input("n1: "))
b =int(input("n2: "))
c =int(input("n3: "))
d =int(input("n4: "))
tupla = (a,b,c,d)
n9 = tupla.count(9)
print(f'o numero 9 apareceu {n9} vezes')
for i in tupla:
  if i ==3:
    print(f'o numero 3 apareceu primeiramente na posição  {tupla.index(3)}')
  if i==3 not in tupla:

    print("não existe numero 3 na tupla")
tupla2 = tuple(i for i in tupla if i%2==0)
print(f' os numeros pares são {tupla2}')


n1: 2
n2: 3
n3: 8
n4: 9
o numero 9 apareceu 1 vezes
o numero 3 apareceu primeiramente na posição  1
 os numeros pares são (2, 8)


76: Crie um programa que tenha uma tupla única com nomes de produtos e seus respectivos preços, na sequência. No final, mostre uma listagem de preços, organizando os dados em forma tabular.

In [ ]:
tupla =('maracuja',12,'maça',10,'banana',15,'laranja',20,'abacaxi',25,'abobora',30,'abacate',35,'goiaba',7)
print('FRUTAS')
for i in tupla:
  if type(i)==str:
    print(f'{i}:{tupla[tupla.index(i)+1]}')


FRUTAS
maracuja:12
maça:10
banana:15
laranja:20
abacaxi:25
abobora:30
abacate:35
goiaba:7


77: Crie um programa que tenha uma tupla com várias palavras (não usar acentos). Depois disso, você deve mostrar, para cada palavra, quais são as suas vogais

In [ ]:
tupla =('maracuja','maça','banana','laranja','abacaxi','abobora','abacate','goiaba','wwwwq','pneumoultramicroscopicsilicovulcanoconiose')
l = []
for palavra in tupla:
  v = 0
  for i in palavra:
    if i in 'aeiou':
      v+=1
      l.append(i)
  print(f'a palavra {palavra} tem {v} vogais são elas: {l}')
  l.clear()





a palavra maracuja tem 4 vogais são elas: ['a', 'a', 'u', 'a']
a palavra maça tem 2 vogais são elas: ['a', 'a']
a palavra banana tem 3 vogais são elas: ['a', 'a', 'a']
a palavra laranja tem 3 vogais são elas: ['a', 'a', 'a']
a palavra abacaxi tem 4 vogais são elas: ['a', 'a', 'a', 'i']
a palavra abobora tem 4 vogais são elas: ['a', 'o', 'o', 'a']
a palavra abacate tem 4 vogais são elas: ['a', 'a', 'a', 'e']
a palavra goiaba tem 4 vogais são elas: ['o', 'i', 'a', 'a']
a palavra wwwwq tem 0 vogais são elas: []
a palavra pneumoultramicroscopicsilicovulcanoconiose tem 19 vogais são elas: ['e', 'u', 'o', 'u', 'a', 'i', 'o', 'o', 'i', 'i', 'i', 'o', 'u', 'a', 'o', 'o', 'i', 'o', 'e']


78: Faça um programa que leia 5 valores numéricos e guarde-os em uma lista. No final, mostre qual foi o maior e o menor valor digitado e as suas respectivas posições na lista.

In [ ]:
ass = []
for i in range(5):
  a =int(input("digite um numero: "))
  ass.append(a)
print(f' o maior valor da lista é  {max(ass)} e o menor é {min(ass)}')

digite um numero: 1
digite um numero: 2
digite um numero: 3
digite um numero: 4
digite um numero: 5
 o maior valor da lista é  5 e o menor é 1


79: Crie um programa onde o usuário possa digitar vários valores numéricos e cadastre-os em uma lista. Caso o número já exista lá dentro, ele não será adicionado. No final, serão exibidos todos os valores únicos digitados, em ordem crescente.

In [ ]:
l = []
while True:
  n  = int(input("digite um numero: "))
  if n not in l:
    l.append(n)
  elif n in l:
    print("esse numero ja existe na lista!")
  c =str(input("deseja continuar: [S/N]: "))
  if c.upper() =='N':
    break

print(f'lista em ordem crescente{sorted(l)}')


digite um numero: 1
deseja continuar: [S/N]: s
digite um numero: 2
deseja continuar: [S/N]: s
digite um numero: 1
esse numero ja existe na lista!
deseja continuar: [S/N]: s
digite um numero: 4
deseja continuar: [S/N]: s
digite um numero: 3
deseja continuar: [S/N]: s
digite um numero: 78
deseja continuar: [S/N]: s
digite um numero: 65
deseja continuar: [S/N]: s
digite um numero: 43
deseja continuar: [S/N]: s
digite um numero: 21
deseja continuar: [S/N]: s
digite um numero: 23
deseja continuar: [S/N]: s
digite um numero: -3
deseja continuar: [S/N]: n
[-3, 1, 2, 3, 4, 21, 23, 43, 65, 78]


80: Crie um programa onde o usuário possa digitar cinco valores numéricos e cadastre-os em uma lista, já na posição correta de inserção (sem usar o sort()). No final, mostre a lista ordenada na tela.

In [ ]:
l = []
organizada = []
for i in range(5):
  n  = int(input("digite um numero: "))
  l.append(n)
for c in range(min(l),max(l)+1):
    if c in l:
      organizada.append(c)
print(f' lista desorganizada: {l}')
print(f'lista organizada: {organizada}')

digite um numero: 3
digite um numero: 43
digite um numero: 2
digite um numero: 1
digite um numero: 8
 lista desorganizada: [3, 43, 2, 1, 8]
lista organizada: [1, 2, 3, 8, 43]


81: Crie um programa que vai ler vários números e colocar em uma lista. Depois disso, mostre:
A) Quantos números foram digitados.
B) A lista de valores, ordenada de forma decrescente.
C) Se o valor 5 foi digitado e está ou não na lista.

In [ ]:
l = []
while True:
  n  = int(input("digite um numero: "))
  l.append(n)
  c = str(input("deseja continuar: [S/N]"))
  if c.upper() =='N':
    break
print(f' foram digitados {len(l)} numeros')
b = sorted(l,reverse=True)
print(f' ordenando a lista na ordem descrescente temos: {b}')
if 5 in l:
  print(f'o numero 5 existe na lista na posição: {b.index(5)}')
else:
  print('o numero 5 não existe na lista ')

digite um numero: 12
deseja continuar: [S/N]s
digite um numero: 21
deseja continuar: [S/N]s
digite um numero: 5
deseja continuar: [S/N]s
digite um numero: 34
deseja continuar: [S/N]s
digite um numero: 65
deseja continuar: [S/N]s
digite um numero: 7
deseja continuar: [S/N]n
 foram digitados 6 numeros
 ordenando a lista na ordem descrescente temos: [65, 34, 21, 12, 7, 5]
o numero 5 existe na lista na posição: 5


82: Crie um programa que vai ler vários números e colocar em uma lista. Depois disso, crie duas listas extras que vão conter apenas os valores pares e os valores ímpares digitados, respectivamente. Ao final, mostre o conteúdo das três listas geradas.


In [ ]:
l = []
pares = []
impares = []
while True:
  n = int(input("digite um numero: "))
  l.append(n)
  c = str(input("deseja continuar: [S/N]"))
  if c in 'Nn':
    break
for i in l:
    if i%2==0:
      pares.append(i)
    else:
      impares.append(i)
print(f' a lista gerada é {l} ela possui {len(l)}')
print(f' lista de pares: {pares}')
print(f'lista de impares: {impares}')

digite um numero: 23
deseja continuar: [S/N]s
digite um numero: 11
deseja continuar: [S/N]s
digite um numero: 12
deseja continuar: [S/N]s
digite um numero: 78
deseja continuar: [S/N]s
digite um numero: 45
deseja continuar: [S/N]s
digite um numero: 1
deseja continuar: [S/N]n
 a lista gerada é [23, 11, 12, 78, 45, 1] ela possui 6
 lista de pares: [12, 78]
lista de impares: [23, 11, 45, 1]


83: Crie um programa onde o usuário digite uma expressão qualquer que use parênteses. Seu aplicativo deverá analisar se a expressão passada está com os parênteses abertos e fechados na ordem correta.

In [ ]:
a = str(input("digite uma expressão matematica: "))
b  = a.count('(')
c = a.count(')')
if b==c and '()' not in a:
  print('sua expresão matematica é valida')
else:
  print("expressão matematica invalida ")

digite uma expressão matematica: (1 + 1 * (50 / ()(23 - 2))
expressão matematica invalida 


In [ ]:
dados = [['matheus',24],['lucas',20],['antonio',21]]
for p in dados:
  print(f'{p[0]}:{p[1]+30}')
print(f'essa sera a idade de todos eles em {2024+30}')

matheus:54
lucas:50
antonio:51
essa sera a idade de todos eles em 2054


84: Faça um programa que leia nome e peso de várias pessoas, guardando tudo em uma lista. No final, mostre:
A) Quantas pessoas foram cadastradas.
B) Uma listagem com as pessoas mais pesadas.
C) Uma listagem com as pessoas mais leves.

In [ ]:
l = []
p = []
while True:
  l.append(str(input("digite seu nome: ")))
  p.append(float(input("digite seu peso: ")))
  c = str(input('deseja continuar [S/N]'))
  if c in 'Nn':
    break
  if c in 'Ss':
    print('dados cadastrados')

print(f'foram cadastradas {len(l)} pessoas.')
print(f'a pessoa mais pesada é {max(p)}')
print(f'a pessoa mais leve é {min(p)}')



digite seu nome: matheus
digite seu peso: 82
deseja continuar [S/N]s
dados cadastrados
digite seu nome: lucas
digite seu peso: 78
deseja continuar [S/N]s
dados cadastrados
digite seu nome: nath
digite seu peso: 50
deseja continuar [S/N]n
foram cadastradas 3 pessoas.
a pessoa mais pesada é 82.0
a pessoa mais leve é 50.0


85: Crie um programa onde o usuário possa digitar sete valores numéricos e cadastre-os em uma lista única que mantenha separados os valores pares e ímpares. No final, mostre os valores pares e ímpares em ordem crescente.

In [ ]:
pares = []
impares = []
n = []
c = 1
while c<=7:
  a = (int(input("digite um numero: ")))
  n.append(a)
  c+=1
  if a%2==0:
    pares.append(a)
  else:
    impares.append(a)

print(f'a lista com todos os numeros é {n}')
print(f'a lista de pares ordenada é  {sorted(pares)}')
print(f'a lista de impares é {sorted(impares)}')

digite um numero: 2
digite um numero: 3
digite um numero: 6
digite um numero: 9
digite um numero: 67
digite um numero: 43
digite um numero: 55
a lista com todos os numeros é [2, 3, 6, 9, 67, 43, 55]
a lista de pares ordenada é  [2, 6]
a lista de impares é [3, 9, 43, 55, 67]


86: Crie um programa que declare uma matriz de dimensão 3x3 e preencha com valores lidos pelo teclado. No final, mostre a matriz na tela, com a formatação correta.

In [ ]:
w = [1,2,3]
l1 =[1,1,1]
l2 =[2,2,2]
l3 =[3,3,3]

for l,c1 in zip(l1,w):
  a = int(input(f'digite o numero de [{l}][{c1}]: '))
  x.append(a)

for l,c2 in zip(l2,w):
  b = int(input(f'digite o numero de [{l}][{c2}]: '))
  y.append(b)
for l,c3 in zip(l3,w):
  v = int(input(f'digite o numero de [{l}][{c3}]: '))
  z.append(v)

print(f'|{x}|')
print(f'|{y}|')
print(f'|{z}|')


digite o numero de [1][1]: 1
digite o numero de [1][2]: 2
digite o numero de [1][3]: 3
digite o numero de [2][1]: 4
digite o numero de [2][2]: 5
digite o numero de [2][3]: 6
digite o numero de [3][1]: 7
digite o numero de [3][2]: 8
digite o numero de [3][3]: 9
|[1, 2, 3]|
|[4, 5, 6]|
|[7, 8, 9]|


087: Aprimore o desafio anterior, mostrando no final:
A) A soma de todos os valores pares digitados.
B) A soma dos valores da terceira coluna.
C) O maior valor da segunda linha.



In [ ]:
w = [1,2,3]
l1 =[1,1,1]
l2 =[2,2,2]
l3 =[3,3,3]
x = []
y = []
z = []
lista = []
soma =0

for l,c1 in zip(l1,w):
  a = int(input(f'digite o numero de [{l}][{c1}]: '))
  x.append(a)

for l,c2 in zip(l2,w):
  b = int(input(f'digite o numero de [{l}][{c2}]: '))
  y.append(b)

for l,c3 in zip(l3,w):
  v = int(input(f'digite o numero de [{l}][{c3}]: '))
  z.append(v)

lista.append(x)
lista.append(y)
lista.append(z)
for a in lista:
  for i in a:
    if i%2==0:
      soma+=i

print(f'|{x}|')
print(f'|{y}|')
print(f'|{z}|')
print(f' a soma de todos os pares é  {soma}')
print(f'a soma de todos os numeros da terceira coluna é  {x[-1]+y[-1]+z[-1]}')
print(f' o maior valor da segunda linha é {max(y)} ')

digite o numero de [1][1]: 1
digite o numero de [1][2]: 2
digite o numero de [1][3]: 3
digite o numero de [2][1]: 4
digite o numero de [2][2]: 5
digite o numero de [2][3]: 6
digite o numero de [3][1]: 7
digite o numero de [3][2]: 8
digite o numero de [3][3]: 9
|[1, 2, 3]|
|[4, 5, 6]|
|[7, 8, 9]|
 a soma de todos os pares é  20
a soma de todos os numeros da terceira coluna é  18
 o maior valor da segunda linha é 6 


88: Faça um programa que ajude um jogador da MEGA SENA a criar palpites.O programa vai perguntar quantos jogos serão gerados e vai sortear 6 números entre 1 e 60 para cada jogo, cadastrando tudo em uma lista composta.

In [ ]:
import random
l = []
sub = []
jogos = []
n  = int(input("digite quantos jogos vc quer: "))
for i in range(n*6):
  a= random.randint(1,61)
  l.append(a)
for i in l:
  sub.append(i)
  if len(sub)==6:
    for i in sub:
      if sub.count(i)>1:
        print(f'essa sequencia {sub} apresenta repetição de numeros')
        b =random.randint(1,61)
        print(i)
        sub.remove(i)
        print(f'o numero {i}  foi removido')
        print(f'e foi substituido por {b}')
        sub.append(b)

    jogos.append(sub[:])
    sub.clear()

for jogo in jogos:

  print(f'jogo {jogos.index(jogo)+1}: {jogo}')




digite quantos jogos vc quer: 6
essa sequencia [58, 41, 5, 32, 17, 5] apresenta repetição de numeros
5
o numero 5  foi removido
e foi substituido por 11
jogo 1: [58, 41, 32, 17, 5, 11]
jogo 2: [52, 8, 37, 31, 25, 46]
jogo 3: [11, 1, 40, 3, 9, 46]
jogo 4: [30, 46, 1, 7, 10, 38]
jogo 5: [38, 33, 60, 7, 46, 58]
jogo 6: [3, 54, 22, 43, 35, 36]


89: Crie um programa que leia nome e duas notas de vários alunos e guarde tudo em uma lista composta. No final, mostre um boletim contendo a média de cada um e permita que o usuário possa mostrar as notas de cada aluno individualmente


In [ ]:
alunos =[]
c  n1 = float(input('digite sua nota 01: '))
  n2 =  float(input("digite sua nota 02: "))
  media =(n1+n2)/2
  alunos.append([nome,[n1,n2],media])
  c = str(input("deseja continuar: [S/N] "))
  if c in 'Nn':
    print('-=-'*20)
    print('BOLETIM')
    print(f'{"no":<4}-{"ALUNO":<10}{"MEDIA":>8}')
    print('-=-'*20)
    for i in alunos:
      print(f'{alunos.index(i):<4} {i[0]:<10} {i[2]:>8.1f}')
    print('-=-'*20)
    break
  if c in 'ss':
    print('ok')

op = int(input("digite o numero do aluno que vc deseja vizualizar as notas: "))
if op in range(len(alunos)):
  print('-=-'*20)
  print(f'ALUNO: {alunos[op][0]}')
  print('-=-'*20)
  print(f'nota 01: {alunos[op][1][0]}')
  print(f'nota 02: {alunos[op][1][1]}')
  print('-=-'*20)
  print(f'MEDIA: {alunos[op][2]}')
else:
  print('numero invalido')
print('obrigado, volte sempre.')

digite o seu nome: matheus
digite sua nota 01: 7
digite sua nota 02: 8
deseja continuar: [S/N] s
ok
digite o seu nome: betania
digite sua nota 01: 9
digite sua nota 02: 8
deseja continuar: [S/N] s
ok
digite o seu nome: mauro caproni
digite sua nota 01: 6
digite sua nota 02: 8
deseja continuar: [S/N] s
ok
digite o seu nome: cidinha rocha alves
digite sua nota 01: 7.8
digite sua nota 02: 9.8
deseja continuar: [S/N] n
-=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=-
BOLETIM
no  -ALUNO        MEDIA
-=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=-
0    matheus         7.5
1    betania         8.5
2    mauro caproni      7.0
3    cidinha rocha alves      8.8
-=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=-
digite o numero do aluno que vc deseja vizualizar as notas: 3
-=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=-
ALUNO: cidinha rocha alves
-=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=--=-
nota 01: 7.8
nota 02: 9.8
-=--=--=--=--=--=

# DICIONARIOS!!

In [ ]:
classe = []
aluno = {}
for i in range(3):
  aluno['nome']=str(input("digite o seu nome: "))
  aluno['idade'] = int(input("digite sua idade: "))
  aluno['materia'] = str(input("qual é sua materia preferida"))
  classe.append(aluno.copy())
print(classe.keys('nome'))


digite o seu nome: a
digite sua idade: 1
qual é sua materia preferidad
digite o seu nome: b
digite sua idade: 2
qual é sua materia preferidar
digite o seu nome: c
digite sua idade: 2323
qual é sua materia preferidag


AttributeError: 'list' object has no attribute 'items'

In [ ]:
for k in classe:
  for v,w in k.items():
    print(f'{v} = {w}')

nome = matheus
idade = 23
materia = fisica
nome = lucas
idade = 21
materia = matematica
nome = be
idade = 24
materia = geografia


90: Faça um programa que leia nome e média de um aluno, guardando também a situação em um dicionário. No final, mostre o conteúdo da estrutura na tela.

In [ ]:
aluno = {}
aluno['nome'] = str(input("digite seu nome:"))
aluno['media']= float(input("digite sua media: "))
if aluno['media']>=7:
  aluno['situação']='aprovado'
else:
  aluno['situação'] = 'reprovado'
print(aluno)

digite seu nome:lucas
digite sua media: 5
{'nome': 'lucas', 'media': 5.0, 'situação': 'reprovado'}


92: Crie um programa que leia nome, ano de nascimento e carteira de trabalho e cadastre-o (com idade) em um dicionário. Se por acaso a CTPS for diferente de ZERO, o dicionário receberá também o ano de contratação e o salário. Calcule e acrescente, além da idade, com quantos anos a pessoa vai se aposentar.

In [ ]:
trabalhador = {}
trabalhador['nome'] = str(input("digite seu nome: "))
trabalhador['data nascimento'] = int(input("digite sua idade: "))
trabalhador['ctps'] = int(input("digite sua CTPS: "))
if trabalhador['ctps'] !=0:
  trabalhador['ano de contratação'] = int(input("digite o ano de sua contratação: "))
  trabalhador['salario'] = float(input("digite o seu salario: "))
trabalhador['tempo de serviço']=2024 - trabalhador['ano de contratação']
a = 30
trabalhador['aposentadoria'] = 2024 +(a-trabalhador['tempo de serviço'])
for a,b in trabalhador.items():
  print(f'{a} tem valor {b}')


#print('com base nos dados fornecidos  o trabalhador', trabalhador['nome'],' irá se aposentar em',trabalhador['aposentadoria'])


digite seu nome: matheus rocha
digite sua idade: 24
digite sua CTPS: 2334342323
digite o ano de sua contratação: 2019
digite o seu salario: 1234
nome tem valor matheus rocha
data nascimento tem valor 24
ctps tem valor 2334342323
ano de contratação tem valor 2019
salario tem valor 1234.0
tempo de serviço tem valor 5
aposentadoria tem valor 2049


93: Crie um programa que gerencie o aproveitamento de um jogador de futebol. O programa vai ler o nome do jogador e quantas partidas ele jogou. Depois vai ler a quantidade de gols feitos em cada partida. No final, tudo isso será guardado em um dicionário, incluindo o total de gols feitos durante o campeonato.

In [ ]:
nome,qtd_part,gols=0,0,0
def jogador():
  nome =str(input('digite seu nome: '))
  qtd_part = int(input("digite a qtd de partidas que vc jogou "))
  gols = int(input("digite a qtd de gols feitos: "))
  dic ={'nome':nome,'partidas':qtd_part,'gols':gols}
  return dic
jogador = jogador()
print('-='*20)
for k,v in jogador.items():
  print(f'{k} ---- {v}')



digite seu nome: matheus
digite a qtd de partidas que vc jogou 70
digite a qtd de gols feitos: 45
-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
nome ---- matheus
partidas ---- 70
gols ---- 45
{'nome': 'matheus', 'partidas': 70, 'gols': 45}


94: Crie um programa que leia nome, sexo e idade de várias pessoas, guardando os dados de cada pessoa em um dicionário e todos os dicionários em uma lista. No final, mostre:
A) Quantas pessoas foram cadastradas
B) A média de idade
C) Uma lista com as mulheres
D) Uma lista de pessoas com idade acima da média

In [ ]:
d = {}
l = []
acimamedia =[]
f = []
soma = 0
c=0
while True:
  d[f'nome{c}'] = str(input("digite seu nome: "))
  d[f'idade_{c}'] = int(input("digite sua idade: "))
  d[f'sexo_{c}'] = str(input("digite seu sexo: "))
  if d[f'sexo_{c}'] == 'MULHER' or d[f'sexo_{c}'] == 'mulher':
    f.append(d[f'nome{c}'])
  l.append(d)
  c+=1
  a = int(input("deseja continuar: [1-s/2-n]"))
  if a==2:
    break

print(l)
#qtd de pessoas cadastradas na lista
print(len(l))
# media de idade:
for i in range(len(l)):
  soma+=l[i][f'idade_{i}']
  media = soma/len(l)
print(f'foram cadastradas {len(l)} pessoas.')
print(f'a media de idade é {media:.2f} anos.')
print(f'a lista de mulheres é {f}')

for id in range(len(l)):
    if d[f'idade_{id}']>media:
      acimamedia.append(d[f'nome{id}'])
print(f'lista com as pessoas que possui a idade acima da media geral: {acimamedia}')




digite seu nome: lara eridan
digite sua idade: 18
digite seu sexo: mulher
deseja continuar: [1-s/2-n]1
digite seu nome: matheus
digite sua idade: 24
digite seu sexo: homem
deseja continuar: [1-s/2-n]1
digite seu nome: antonio
digite sua idade: 21
digite seu sexo: homem
deseja continuar: [1-s/2-n]2
[{'nome0': 'lara eridan', 'idade_0': 18, 'sexo_0': 'mulher', 'nome1': 'matheus', 'idade_1': 24, 'sexo_1': 'homem', 'nome2': 'antonio', 'idade_2': 21, 'sexo_2': 'homem'}, {'nome0': 'lara eridan', 'idade_0': 18, 'sexo_0': 'mulher', 'nome1': 'matheus', 'idade_1': 24, 'sexo_1': 'homem', 'nome2': 'antonio', 'idade_2': 21, 'sexo_2': 'homem'}, {'nome0': 'lara eridan', 'idade_0': 18, 'sexo_0': 'mulher', 'nome1': 'matheus', 'idade_1': 24, 'sexo_1': 'homem', 'nome2': 'antonio', 'idade_2': 21, 'sexo_2': 'homem'}]
3
foram cadastradas 3 pessoas.
a media de idade é 21.00 anos.
a lista de mulheres é ['lara eridan']
lista com as pessoas que possui a idade acima da media geral: ['matheus']


95: Aprimore o desafio 93 para que ele funcione com vários jogadores, incluindo um sistema de visualização de detalhes do aproveitamento de cada jogador.

In [ ]:
jogadores = {}
c = 0
l = []
while True:
  jogadores[f'nome'] = str(input("digite seu nome: "))
  jogadores[f'qtd_part']=int(input("digite s qtd de partidas: "))
  jogadores[f'gols'] = int(input("digite a qtd de gols:"))
  c+=1
  l.append(jogadores.copy())
  w = int(input('deseja continuar: 1-S/2-N'))
  if w==2:
    break
print(l)


digite seu nome: qweqew
digite s qtd de partidas: 123
digite a qtd de gols:321
deseja continuar: 1-S/2-N1
digite seu nome: qwersdgf
digite s qtd de partidas: 3212
digite a qtd de gols:3211
deseja continuar: 1-S/2-N1
digite seu nome: asdsfc
digite s qtd de partidas: 3213
digite a qtd de gols:2121
deseja continuar: 1-S/2-N2
[{'nome': 'qweqew', 'qtd_part': 123, 'gols': 321}, {'nome': 'qwersdgf', 'qtd_part': 3212, 'gols': 3211}, {'nome': 'asdsfc', 'qtd_part': 3213, 'gols': 2121}]


96: Faça um programa que tenha uma função chamada área(), que receba as dimensões de um terreno retangular (largura e comprimento) e mostre a área do terreno.

In [ ]:
def p():
  lar =float(input("digite a largura do terreno: "))
  comp = float(input("digite o comprimento do terreno: "))
  area(lar,comp)


def area(a,b):
  c =a*b
  print(f'a area do terreno é {c}')

p()


digite a largura do terreno: 35
digite o comprimento do terreno: 654
a area do terreno é 22890.0


97: Faça um programa que tenha uma função chamada escreva(), que receba um texto qualquer como parâmetro e mostre uma mensagem com tamanho adaptável.

Ex:
escreva('Olá, Mundo!')
Saída:
~~~~~~~~~
 Olá, Mundo!
~~~~~~~~~

In [ ]:
def escreva(a):
  print('-'*20)
  print(a)
  print('-'*20)
escreva('MATHEUS ROCHA NEMER ')

--------------------
MATHEUS ROCHA NEMER 
--------------------


98: Faça um programa que tenha uma função chamada contador(), que receba três parâmetros: início, fim e passo. Seu programa tem que realizar três contagens através da função criada:

a) de 1 até 10, de 1 em 1
b) de 10 até 0, de 2 em 2
c) uma contagem personalizada

In [ ]:

def de1a10():

  print('contagem de 1 a 10:')
  import time
  for i in range(1,11):
    time.sleep(1)
    print(i,end=',')

def de10a0():

  print('contagem de 10 a 0:')
  for i in range(10,-1,-2):
    time.sleep(1)
    print(i,end=',')

def cont(a,b,c):
  print('contagem personalizada:')
  for i in range(a,b,c):
    time.sleep(1)
    print(i,end=',')



de1a10()

de10a0()

cont(-13,20,2)

contagem de 1 a 10:
1,2,3,4,5,6,7,8,9,10,contagem de 10 a 0:
10,8,6,4,2,0,contagem personalizada:
-13,-11,-9,-7,-5,-3,-1,1,3,5,7,9,11,13,15,17,19,

99: Faça um programa que tenha uma função chamada maior(), que receba vários parâmetros com valores inteiros. Seu programa tem que analisar todos os valores e dizer qual deles é o maior.

In [ ]:
def maior(*num):
  maior  =0
  for i in num:
    if i>maior:
      maior = i
  print(f'foram recebidos {len(num)} numeros')
  print(f'o maior valor recebido é {maior}')
maior(2,4,32,2,12,43,34,56,32,2,1,1,111,34,434,543,90)

foram recebidos 17 numeros
o maior valor recebido é 543


 100: Faça um programa que tenha uma lista chamada números e duas funções chamadas sorteia() e somaPar(). A primeira função vai sortear 5 números e vai colocá-los dentro da lista e a segunda função vai mostrar a soma entre todos os valores pares sorteados pela função anterior.

In [ ]:
def sorteia():
  lista = []
  import random
  for i in  range(5):
    b = random.randint(1,30)
    lista.append(b)
  print(f'a lista sorteada é  {lista}')
  soma(lista)
def soma(lista):
    print(f'a soma de todos os valores da lista é {sum(lista)}')
sorteia()


a lista sorteada é  [27, 12, 25, 5, 19]
a soma de todos os valores da lista é 88


101: Crie um programa que tenha uma função chamada voto() que vai receber como parâmetro o ano de nascimento de uma pessoa, retornando um valor literal indicando se uma pessoa tem voto NEGADO, OPCIONAL e OBRIGATÓRIO nas eleições.

In [ ]:
def voto(a):
  c = 2024-a
  if c>=18 and c<70:
    print(' voto obrigatorio')
  elif c>=70:
    print('voto opcional')
  else:
    print('voto negado')

ano = int(input("digite seu ano de nascimento: "))
voto(ano)

digite seu ano de nascimento: 1941
voto opcional


102: Crie um programa que tenha uma função fatorial() que receba dois parâmetros: o primeiro que indique o número a calcular e outro chamado show, que será um valor lógico (opcional) indicando se será mostrado ou não na tela o processo de cálculo do fatorial.

In [ ]:
def fat(a,show):
  f = 1
  if show =='sim':
    for c in range(a,1,-1):
      f*=c
      print(f'{f}',end=' ')
  elif show =='não':
    for c in range(a,0,-1):
      f*=c
    print(f'valor final fatorial de {a} é {f}')
n = int(input("digite o numero que vc quer o fatorial: "))
p =str(input("vc deseja ver o precesso: sim ou não"))
fat(n,p)

digite o numero que vc quer o fatorial: 8
vc deseja ver o precesso: sim ou nãosim
8 56 336 1680 6720 20160 40320 

103: Faça um programa que tenha uma função chamada ficha(), que receba dois parâmetros opcionais: o nome de um jogador e quantos gols ele marcou. O programa deverá ser capaz de mostrar a ficha do jogador, mesmo que algum dado não tenha sido informado corretamente.



In [ ]:
def ficha(nome='jogador',gols = 0):
  print('FICHA DO JOGADOR')
  print(f'NOME:{nome}     GOLS:{gols}')
nome = str(input("digite o nome do jogador: "))
gols = int(input("digite a qtd de gols que o jogador fez: "))
ficha(nome,gols)


digite o nome do jogador: 
digite a qtd de gols que o jogador fez: 


ValueError: invalid literal for int() with base 10: ''

 104: Crie um programa que tenha a função leiaInt(), que vai funcionar de forma semelhante 'a função input() do Python, só que fazendo a validação para aceitar apenas um valor numérico.
Ex: n = leiaInt('Digite um n: ')

In [ ]:
def leiaint():
  a = int(input("digite um numero: "))
  print(f' o numero lido é {a}')
leiaint()

digite um numero: 23
 o numero lido é 23


105: Faça um programa que tenha uma função notas() que pode receber várias notas de alunos e vai retornar um dicionário com as seguintes informações:

- Quantidade de notas
- A maior nota
- A menor nota
- A média da turma
- A situação (opcional)

In [ ]:
def notas():
  d = {'quantidade de notas':'','maior nota':'','menor nota':'','media da turma':''}
  a = []
  notas = []
  while True:
    aluno = str(input("digite seu nome: "))
    n = float(input("digite sua nota: "))
    a.append(aluno)
    notas.append(n)
    s = int(input("deseja continuar: 1 - sim 2 - não"))
    if s==1:
      print('ok')
    elif s==2:
      d['quantidade de notas'] =  len(a)
      d['maior nota'] = max(notas)
      d['menor nota'] = min(notas)
      d['media da turma'] =(sum(notas))/len(a)
      print(d)

      break

notas()



106: Faça um mini-sistema que utilize o Interactive Help do Python. O usuário vai digitar o comando e o manual vai aparecer. Quando o usuário digitar a palavra 'FIM', o programa se encerrará. Importante: use cores.

In [ ]:

while True:
  a = str(input("Função ou biblioteca? "))
  help(a)
  if a=='sair':
    break
print(fore.red+'ate logo')

ModuleNotFoundError: No module named 'color'

In [ ]:
for i in l:
  print('-=-'*10)
  print(f'JOGADOR   {l.index(i)}')
  print('-=-'*10)
  for k,v in i.items():
    print(f'{k} ----{v}')


-=--=--=--=--=--=--=--=--=--=-
JOGADOR   0
-=--=--=--=--=--=--=--=--=--=-
nome ----qweqew
qtd_part ----123
gols ----321
-=--=--=--=--=--=--=--=--=--=-
JOGADOR   1
-=--=--=--=--=--=--=--=--=--=-
nome ----qwersdgf
qtd_part ----3212
gols ----3211
-=--=--=--=--=--=--=--=--=--=-
JOGADOR   2
-=--=--=--=--=--=--=--=--=--=-
nome ----asdsfc
qtd_part ----3213
gols ----2121


In [ ]:
l =[7,5,3,2,12,23]
def dobra(list):
  nova = []
  for i in list:
    nova.append(i*2)
  print(nova)
dobra(l)


[14, 10, 6, 4, 24, 46]


In [ ]:
print(len(l))

3


In [ ]:
'''a ={'nome':'matheus','idade1':24,'sexo1':'m','nome1': 'lucas','idade2':20,'sexo2':'m','nome2': 'antonio','idade3':21,'sexo3':'m'}
for i in a.keys():
  print(f'a chave {i} tem como valor {a[i]}')'''
b ={}
l =[]
c = 0
while True:
  b[f'nome{c}'] = str(input("digite seu nome: "))
  b[f'idade_{c}'] = int(input("digite sua idade: "))
  b[f'sexo_{c}'] = str(input("digite seu sexo: "))
  c+=1
  l.append()
  a = int(input("deseja continuar: [1-s/2-n]"))
  if a==2:
    break

print(b)

digite seu nome: matheus
digite sua idade: 25
digite seu sexo: homem
deseja continuar: [1-s/2-n]1
digite seu nome: lara
digite sua idade: 18
digite seu sexo: mulher
deseja continuar: [1-s/2-n]2
{'nome0': 'matheus', 'idade_0': 25, 'sexo_0': 'homem', 'nome1': 'lara', 'idade_1': 18, 'sexo_1': 'mulher'}


In [ ]:
'''for i in range(2):

  a ={'nome':str(input("digite seu nome: ")),'salario':float(input("digite seu salario: "))}
print(a)'''
d ={'nome':'matheus','salario':1442,'nome':'lucas','salario':1567}
d.values()

dict_values(['lucas', 1567])

In [ ]:
'''a1 = []
a2 = []
a3 = []
for i in range(3):
  print(i)
  if i==0:
    a1.append(nome)
    a1.append(salario)
  if i==1:'''
a = [1,2,3,4,5,6,7,8]
for c in a[::2]:
  print(c)

1
3
5
7


In [ ]:
d ={}
l = []
maior = 0
for i in range(1):
  d =  {}

  d["nome"] = input("digite seu nome: ")
  d["função"] = input("digite sua função: ")
  d["salario"] = float(input("digite seu salario "))
  l.append(d)
print(l)
for i in l:
  print(i)


'''for p in l:
  if p['salario']==maior:
    print(f'maior salario: {l['salario']}')
    print(f'o funcionario com o maior salario é {l['nome']}')'''



digite seu nome: matheus
digite sua função: qwer2
digite seu salario 12345
[{'nome': 'matheus', 'função': 'qwer2', 'salario': 12345.0}]
{'nome': 'matheus', 'função': 'qwer2', 'salario': 12345.0}


"for p in l:\n  if p['salario']==maior:\n    print(f'maior salario: {l['salario']}')\n    print(f'o funcionario com o maior salario é {l['nome']}')"

In [ ]:
a ={'a':1,'b':[2,1]}

c =int(input())
a['c']=c
print(a)
for i in a.keys():
  if i=='b':
    print(a[i][0])

11
{'a': 1, 'b': [2, 1], 'c': 11}
2


In [ ]:
a ={'a':{'8h00 - matheus rocha '},'b': {'17h00 - lucas'}}
print(a['a'])

{'8h00 - matheus rocha '}


In [ ]:
lista_pacientes ={}
consultas= {}
segundaof,quartaof,sextaof ={},{},{}
terça,quinta,sabado={},{},{}
oftalmo ={'segundaof':[],'quartaof':[],'sextaof':[]}
endo ={'terça':[],'quinta':[],'sabado':[]}

cardiologista ={}
nutricionista = {}
print('--'*20)
print('MENU PRINCIPAL')
print('--'*20)
d ={}
while True:
    op = int(input("vc é (1)paciente (2)medico (3)sair: "))
    if op==1:
        print('bem vindo usuario')
        nome=str(input("digite seu nome: "))
        d['cpf']=int(input("digite seu cpf: "))
        d['idade']=int(input("digite sua idade: "))
        print(d)
        print('--'*20)
        print('''MENU
              (1) REALIZAR CONSULTA
              (2) VER HISTORICO DE CONSULTAS''')
        print('--'*20)

        menu = int(input("digite sua opção: "))
        if menu==1:
            op_area= int(input('''vc deseja se consultar com qual profissional:
                  (1)Oftalmologista
                  (2)Endocrinologista

                  '''))
            if op_area ==1:
                print('--'*20)
                print('OFTALMOLOGIA')
                print('--'*20)
                op =int(input('''MEDICOS
                    ------------------------------------------------
                      (1)DRA Gabriela marques -segunda/quarta/sexta

                    -------------------------------------------------
                        Digite sua opção: '''))
                if op==1:
                   dia =int(input('''Selecione o dia da sua consulta:
                          (1)SEGUNDA
                          (2)QUARTA
                          (3)SEXTA  '''))
                   if dia!=1 and dia!=2 and dia !=3:
                       print('invalido')
                   if dia==1:
                       hr =int(input(
                           '''selecione o horario:
                          (1)08H00
                          (2)10H00
                          (3)15H00
                          (4)17H00  '''))
                       if hr==1:
                           oftalmo['segundaof']=[nome,'08h00']
                           print('consulta marcada')
                       elif hr==2:
                           oftalmo['segundaof']=[nome,'10h00']
                       elif hr==3:
                           oftalmo['segundaof']=[nome,'15h00']
                           print('consulta marcada')
                       elif hr==4:
                           oftalmo['segundaof']=[nome,'17h00']
                           print('consulta marcada')

                   if dia==2:
                       hr =int(input(
                           '''selecione o horario:
                          (1)08H00
                          (2)10H00
                          (3)15H00
                          (4)17H00  '''))
                       if hr==1:
                           oftalmo['quartaof']=[nome,'08h00']
                           print('consulta marcada')
                       elif hr==2:
                           oftalmo['quartaof']=[nome,'10h00']
                           print('consulta marcada')
                       elif hr==3:
                           oftalmo['quartaof']=[nome,'15h00']
                           print('consulta marcada')
                       elif hr==4:
                           oftalmo['quartaof']=[nome,'17h00']
                           print('consulta marcada')

                   if dia==3:
                       hr =int(input(
                           '''selecione o horario:
                          (1)08H00
                          (2)10H00
                          (3)15H00
                          (4)17H00  '''))
                       if hr==1:
                           oftalmo['sextaof']=[nome,'08h00']
                           print('consulta marcada!')
                       elif hr==2:
                           oftalmo['sextaof']=[nome,'10h00']
                           print('consulta marcada!')
                       elif hr==3:
                           oftalmo['sextaof']=[nome,'15h00']
                           print('consulta marcada!')
                       elif hr==4:
                           oftalmo['sextaof']=[nome,'17h00']
                           print('consulta marcada!')
                           print(oftalmo)

                       if hr not in[1,2,3,4]:
                           print('opção invalida')
            if op_area==2:
                    print('--'*20)
                    print('ENDOCRINOLOGISTA')
                    print('--'*20)
                    op =int(input('''MEDICOS
                    ------------------------------------------------
                      (1)DR: Dante alighieri -terça/quinta/sabado
                    (2) Sair

                    -------------------------------------------------
                        Digite sua opção: '''))
                    if op==1:
                        dia =int(input('''Selecione o dia da sua consulta:
                          (1)TERÇA
                          (2)QUINTA
                          (3)SABADO '''))
                        if dia!=1 and dia!=2 and dia !=3:
                            print('invalido')

                        if dia==1:
                                hr =int(input(
                                    '''selecione o horario:
                                    (1)08H00
                                    (2)10H00
                                    (3)15H00
                                    (4)17H00'''))
                                if hr==1:
                                    endo['terça']=[nome,'08h00']
                                    print('consulta marcada!')
                                    print(endo)
                                elif hr==2:
                                    endo['terça']=[nome,'10h00']
                                    print('consulta marcada!')
                                elif hr==3:
                                    endo['terça']=[nome,'15h00']
                                    print('consulta marcada!')
                                    print(endo)

                                elif hr==4:
                                    endo['terça']=[nome,'17h00']
                                    print('consulta marcada!')
                                    print(endo)
                                    print(terça)
                                if hr not in[1,2,3,4]:
                                    print('opção invalida')
                        if dia==2:
                                hr =int(input(
                                    '''selecione o horario:
                                    (1)08H00
                                    (2)10H00
                                    (3)15H00
                                    (4)17H00'''))
                                if hr==1:
                                    endo['quinta']=[nome,'08h00']
                                    print('consulta marcada!')
                                elif hr==2:
                                    endo['quinta']=[nome,'10h00']
                                    print('consulta marcada!')
                                elif hr==3:
                                    endo['quinta']=[nome,'15h00']
                                    print('consulta marcada!')
                                elif hr==4:
                                    endo['quinta']=[nome,'17h00']
                                    print('consulta marcada!')
                                    print(endo)
                                    print(quinta)
                        if dia==3:
                                hr =int(input(
                                    '''selecione o horario:
                                    (1)08H00
                                    (2)10H00
                                    (3)15H00
                                    (4)17H00'''))
                                if hr==1:
                                    endo['sabado']=[nome,'08h00']
                                    print('consulta marcada!')
                                elif hr==2:
                                    endo['sabado']=[nome,'10h00']
                                    print('consulta marcada!')
                                elif hr==3:
                                    endo['sabado']=[nome,'15h00']
                                    print('consulta marcada!')
                                elif hr==4:
                                    endo['sabado']=[nome,'17h00']
                                    print('consulta marcada!')
                                    print(endo)
                                    print(sabado)
        if menu==2:
                print('VIZUALIZAR HISTORICO')
                for i,j  in zip(endo.values(),oftalmo.values()):
                   print(i,j)
    if op==3:
            break
print('sistema encerrado')

----------------------------------------
MENU PRINCIPAL
----------------------------------------
vc é (1)paciente (2)medico (3)sair: 3
sistema encerrado


In [ ]:
def area(largura,comprimento):
    area = largura*comprimento
    return area

def pot(tipo,ar):
    if tipo ==1:
      p = 15*ar
    elif tipo ==2:
      p = 20*ar
    elif tipo ==3:
      p = 25*ar
    elif tipo ==0:
      p = ar*12
    else:
      print('invalido')
    return p
def lam(p):
    import math
    lt = math.ceil(p/60)

    return lt

def menu():
  tipo = int(input("digite o tipo do comodo: "))
  largura = float(input("digite a largura do comodo: "))
  comprimento =  float(input("digite o comprimento  do comodo:  "))

  x = area(largura,comprimento)
  y = pot(tipo,x)
  l = lam(y)
  print(f'A area total do comodo é {x} \n a potencia total em watts é {y} e o numero de lampadas ideal é {l}')
menu()

digite o tipo do comodo: 2
digite a largura do comodo: 55
digite o comprimento  do comodo:  65
A area total do comodo é 3575.0 
 a potencia total em watts é 71500.0 e o numero de lampadas ideal é 1192


In [ ]:
pares = 0
impares = 0
def var(a):
  pares = 0
  impares = 0
  for i in a:
    if i%2==0:
      pares +=i
    elif i%2!=0:
      impares+=i
  return pares,impares

In [ ]:
def x(a,b):
  return a**b
def a(c,b):

  return x(c,b) +1



# duvidas com dicionarios  


In [ ]:
#
a ={'nome':'matheus','idade1':24,'sexo1':'m','nome1': 'lucas','idade2':20,'sexo2':'m','nome2': 'antonio','idade3':21,'sexo3':'m'}
for i in a.keys():
  print(f'a chave {i} tem como valor {a[i]}')


a chave nome tem como valor matheus
a chave idade1 tem como valor 24
a chave sexo1 tem como valor m
a chave nome1 tem como valor lucas
a chave idade2 tem como valor 20
a chave sexo2 tem como valor m
a chave nome2 tem como valor antonio
a chave idade3 tem como valor 21
a chave sexo3 tem como valor m
